### PR TITLE
[64][PlacementStrategy] Expand and replace Timeout

### PR DESCRIPTION
**Pull request by dhaiducek**
Replace `Timeout` with `MinSuccessTime`, `ProgressDeadline`, and `MaxFailedPercent`.

**Notes:**
- There was a request to rename `SuccessDeadline` to `ProgressDeadline` and, not seeing opposition to it, I made the change

### CODE REVIEW

File: enhancements/sig-architecture/64-placementStrategy/README.md [hunk 1]
```diff
@@ -470,15 +470,28 @@ type RolloutStrategy struct {
...
 	// +kubebuilder:validation:Pattern="^(([0-9])+[h|m|s])|None$"
 	// +kubebuilder:default:=None
 	// +optional
-	Timeout string `json:"timeout,omitempty"`
+  ProgressDeadline string `json:"progressDeadline,omitempty"`
+  // MaxFailures is a percentage of the current rollout that can fail before proceeding to the next rollout.
+	// Default is that no failures are tolerated.
+  // +kubebuilder:validation:Maximum=100
+	// +optional
+  MaxFailedPercent uint8 `json:"maxFailedPercent,omitempty"`
```
Comment (serngawy) [inline enhancements/sig-architecture/64-placementStrategy/README.md:494]: I believe the name should be **maxFailures**  and better to have type as intstr.IntOrString to allow percentage and numbers with validation as below.
`// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"`

Comment (dhaiducek) [reply enhancements/sig-architecture/64-placementStrategy/README.md]: Would that be similar to `ClustersPerDecisionGroup` where it's either a number of clusters or a percentage of clusters? If so, that sounds good to me.

Comment (serngawy) [reply enhancements/sig-architecture/64-placementStrategy/README.md]: yes, similar to ClustersPerDecisionGroup

**Comment by serngawy** [comment]
Sounds good
/LGTM

**Pull request merged**